compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmValueGen.java

```diff
@@ -1287,7 +1290,7 @@ private void createRecordGetKeysMethod(ClassWriter cw, Map<String, BField> field
             Label ifNotPresent = new Label();
 
             // If its an optional field, generate if-condition to check the presense of the field.
-            String fieldName = optionalField.name.value;
+            String fieldName = IdentifierEncoder.decodeIdentifiers(optionalField.name.value);
```

Comment:
check if we need encoding for these identifiers if we don't use the encoded identifiers